cirq/contrib/noise_models/noise_models.py

```diff
@@ -142,3 +144,51 @@ def noisy_moment(self, moment: 'cirq.Moment',
                 moment,
                 ops.Moment(self.qubit_noise_gate(q) for q in system_qubits)
             ]
+
+
+class PerQubitDepolarizingNoiseModel(devices.NoiseModel):
+    """DepolarizingNoiseModel which allows depolarization probabilities to be
+    specified separately for each qubit.
+
+    Similar to depol_prob in DepolarizingNoiseModel, depol_prob_map should map
+    Qids in the device to their depolarization probability.
+    """
+
+    def __init__(
+            self,
+            depol_prob_map: Dict['cirq.Qid', float],
+    ):
+        """A depolarizing noise model with variable per-qubit noise.
+
+        Args:
+            depol_prob_map: Map of depolarizing probabilities for each qubit.
+        """
+        for qubit, depol_prob in depol_prob_map.items():
+            value.validate_probability(depol_prob, f'depol prob of {qubit}')
+        self.depol_prob_map = depol_prob_map
+
+    def noisy_moment(self, moment: 'cirq.Moment',
+                     system_qubits: Sequence['cirq.Qid']):
+        if _homogeneous_moment_is_measurements(moment):
+            return moment
+        else:
+            return [
+                moment,
+                ops.Moment(
+                    ops.DepolarizingChannel(self.depol_prob_map[q])(q)
+                    for q in system_qubits)
+            ]
+
+
+def simple_noise_from_calibration_metrics(eng: engine.Engine, processor_id: str
+                                         ) -> devices.NoiseModel:
```

Comment:
I think this should take a calibration object instead of an engine object.  Note that a few of the problems you specified in the description would be solved by specifying a calibration object instead.  I think doing this online only saves one line of code and is not necessarily what you want.  Like, I don't know that you want your noise model to suddenly shift if we upload new calibrations.  Also, you wouldn't be able to easily go back and run a consistent noise model (i.e. I want to run the same thing as yesterday).

Reply:
Makes sense - done. The test still uses the 'retrieve Calibration from (mock) engine' step, partly because it required the least change and partly because it's a decent example for users.